vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(admiral)
+link <- function(text, url) {
+  return(
+    paste0(
+      "[", text, "]",
+      "(", url, ")"
+    )
+  )
+}
+dyn_link <- function(text,
+                     base_url,
+                     relative_url = "",
+                     # Change to TRUE when admiral adopts multiversion docs
+                     is_multiversion = FALSE,
+                     multiversion_default_ref = "main") {
+  url <- paste(base_url, relative_url, sep = "/")
+  if (is_multiversion) {
+    url <- paste(
+      base_url,
+      Sys.getenv("BRANCH_NAME", multiversion_default_ref),
+      relative_url,
+      sep = "/"
+    )
+  }
+  return(link(text, url))
+}
+# Other variables
+admiral_homepage <- "https://pharmaverse.github.io/admiral"
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes creating an `ADTR` (Tumor Results) ADaM with common
+oncology parameters based on RECIST v1.1.
+
+The main part in programming a tumor results dataset is the calculation of the
+sum of diameters of all target lesions (lymph nodes & non-lymph nodes), the
+calculation of nadir, change in & percentage change from baseline, and the
+analysis flags that could be setup for reporting.  The tumor results data could
+be set up for investigator and/or Independent review facility (IRF)/Blinded
+Independent Central Review (BICR) data.
+
+The source dataset used will depend on each company,  this could be solely the
+TR domain or you may merge `TU` with `TR` to get additional data variables or to
+ensure that you're processing the same lesions as collected.
+
+Individual sum of diameter for each target lesion (from `TR`) is required to
+calculate the sum of all target lesions, this data could be taken directly from
+`TR` or additional parameters could be created in ADTR (or similar) depending on
+the additional processing required (e.g. imputation of dates, re-labeling of
+visits) and your company specifications.
+
+NEED TO UPDATE THIS PART ONCE TEMPLATE CODE IS IN. The majority of the functions
+used here exist from `{admiral}`, except for the `tte_sources` helper object,
+provided as an example from `{admiralonco}`.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless
+otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages. (This is to be
+reviewed/updated once template code is written)
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(dplyr)
+library(admiral.test)
+library(lubridate)
+library(stringr)
+library(admiralonco)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Merge TR with TU and derive new variables](#merge)
+* [Pre-processing of input records](#input)
+* [Derive Parameters (`TLDn`, `TLPn`)](#parameters)
+* [Derive Parameters (TSD) and ANL01FL](#tsd)
```

Comment:
**[From Catherine Jin via Stephen Gormley]**

`[Derive Parameters (TSD) and ANL01FL]` is for the general data structure:

1.	 CDISC has instruction to have PARAMCD and PARAM in pair with 1-1 mapping, current PARAMCD=’TSD’ is general on tumor evaluator while PARAM ‘Target Lesions Sum of Diameters by Investigator’ indicates investigator, which could cause issue when company needs both INV and ICR in ADTR.

2.	 PARAM and PARCATx is many-one relation, and only one PARCATx should associate with a given PARAMCD.
So I suggest introducing PARQUAL (Breast Cancer TAUG and Prostate Cancer TAUG both used PARQUAL for evaluator) instead of PARCAT2 and remove “by Investigator” in PARAM.

Reply:
I noticed that ADRS also has same conventions. How about keeping TSD for investigator, then for IRF, a new parameter name e.g. TSDIRF or whatever company convention is?

We have the same scenario is ADRS, was this discussed before? Anyone recall?